src/Evolution/Systems/ScalarWave/Initialize.hpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/EagerMath/Norms.hpp"
+#include "DataStructures/Variables.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Evolution/Initialization/DiscontinuousGalerkin.hpp"
+#include "Evolution/Initialization/Evolution.hpp"
+#include "Evolution/Systems/ScalarWave/Characteristics.hpp"
+#include "Evolution/Systems/ScalarWave/Constraints.hpp"
+#include "Evolution/Systems/ScalarWave/System.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+
+namespace ScalarWave {
+namespace Actions {
+/// \ingroup InitializationGroup
+/// \brief Initialize items related to constraints of the ScalarWave system
+///
+/// We add both constraints and the constraint damping parameter to the
+/// evolution databox.
+///
+/// DataBox changes:
+/// - Adds:
+///   * `ScalarWave::Tags::ConstraintGamma2`
+///   * `ScalarWave::Tags::OneIndexConstraint<Dim>`
+///   * `ScalarWave::Tags::TwoIndexConstraint<Dim>`
+///   * `::Tags::PointwiseL2Norm<ScalarWave::Tags::OneIndexConstraint<Dim>>`
+///   * `::Tags::PointwiseL2Norm<ScalarWave::Tags::TwoIndexConstraint<Dim>>`
+/// - Removes: nothing
+/// - Modifies: nothing
+///
+template <size_t Dim>
+struct InitializeConstraints {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using compute_tags =
+        db::AddComputeTags<ScalarWave::Tags::ConstraintGamma2Compute,
```

Comment:
After looking at the location of the compute function, we wondered why it is located in Characteristcs.hpp. It should be moved to Constraints.hpp, like it is done for the curved scalar wave.

Furthermore the ComputeTag just sets Gamma2 to zero. Its value should be set from a parameter in the input file. (Does not have to be in this PR)

Reply:
Moved the compute item to `Constraints.hpp`

We can't set `Gamma2` from input file as in general it can have a non-trivial spatial profile.

Reply:
It is true, that `Gamma2` can have a non-trivial spatial profile. But even a non-trivial profile will have to be set by some parameters, e.g. in the input file.

The constant `Gamma2` would be a simple class of different possible profiles.
I think it is important to give the user at least some control over the damping, instead of turning the damping off by setting it to zero.

Reply:
> I think it is important to give the user at least some control over the damping, instead of turning the damping off by setting it to zero.

I agree with this, and I think this point applies more broadly than this PR since such constraint damping parameters are used in at least 3 evolution systems, least important of which is this one (`ScalarWave`). I have opened issue #2116 which I plan to address outside of this PR, hope that is agreeable with you. Thanks @nilsleiffischer !

Reply:
I think your comment was addressed to @hrueter :)

Reply:
woops!